src/test/kotlin/com/bloock/sdk/record/entity/document/JsonDocumentTest.kt

```diff
@@ -0,0 +1,141 @@
+        val file2 = JsonDocument(JsonDocumentContent(json), JsonDocumentLoadArgs())
+
+        runBlocking {
+
+            file.ready.await()
+            val docPayload = file.getDocPayload()
+            val docPayload2 = file2.getDocPayload()
+            val docProof = file.getDocPayload()
+            val docProof2 = file2.getDocPayload()
+            val docSignatures = file.getDocSignatures()
+            val docSignatures2 = file2.getDocSignatures()
+
+            Assert.assertEquals(docPayload, docPayload2)
+            Assert.assertEquals(docProof, docProof2)
+            Assert.assertTrue(docSignatures?.value!!.isNotEmpty())
+            Assert.assertEquals(docSignatures, docSignatures2)
+        }
+    }
+
+    @Test
+    fun twoIdenticalFilesGeneratesSamePayload() {
+        val file = JsonDocument(JsonDocumentContent(mutableMapOf(Pair("hello", "world"))), JsonDocumentLoadArgs())
+        val file2 = JsonDocument(JsonDocumentContent(mutableMapOf(Pair("hello", "world"))), JsonDocumentLoadArgs())
+
+        runBlocking {
+
+            file.ready.await()
+            val docPayload = file.getDocPayload()
+            val docPayload2 = file2.getDocPayload()
+            val docProof = file.getDocPayload()
+            val docProof2 = file2.getDocPayload()
+            val docSignatures = file.getDocSignatures()
+            val docSignatures2 = file2.getDocSignatures()
+
+            Assert.assertEquals(docPayload, docPayload2)
+            Assert.assertEquals(docProof, docProof2)
+            Assert.assertEquals(docSignatures, docSignatures2)
+        }
+    }
+
+    @Test
+    fun testSetSignature() {
+        val file = JsonDocument(JsonDocumentContent(mutableMapOf(Pair("hello", "world"))), JsonDocumentLoadArgs())
+        val signatures = Signatures(mutableListOf("someSignature", "someSignature2"))
+
+        runBlocking {
+            file.ready.await()
+            file.addSignature(signatures)
+            val docSignatures = file.getDocSignatures()
+            Assert.assertEquals(signatures, docSignatures)
+
+            var src = file.build().await()
+            val file2 = JsonDocument(src, JsonDocumentLoadArgs())
+            file2.ready.await()
+
+            val docSignatures2 = file2.getDocSignatures()
+            Assert.assertEquals(docSignatures2, docSignatures)
+
+        }
+    }
+
+    @Test
+    fun testSetSignatureWithProof() {
+
+        val file = JsonDocument(JsonDocumentContent(mutableMapOf(Pair("hello", "world"))), JsonDocumentLoadArgs())
+        var signatures: MutableList<Any>?
+        signatures = mutableListOf("someSignature", "someSignature2")
+        val proof = SignatureWithProof(
+            Proof(
+                leaves = listOf("leave1"),
+                nodes = listOf("node1"),
+                "deepth",
+                "bitmap"
+            ),
+            signatures = signatures
+        )
+        runBlocking {
+
+            file.ready.await()
+            file.addSignature(proof)
```

Comment:
addSignatures should not take both signature and proof. It must be separated into two functions (addSignatures and setProof)